src/builtins.ts

```diff
@@ -3593,6 +3661,11 @@ export function compileCall(
       let type = evaluateConstantType(compiler, typeArguments, operands, reportNode);
       compiler.currentType = Type.u32;
       if (!type) return module.unreachable();
+      let signatureReference = type.signatureReference;
+      if (type.is(TypeFlags.REFERENCE) && signatureReference !== null) {
```

Comment:
Here, the `!== null` is fine to force a bool on both sides of the logical operator.

Reply:
This is just a comment then?